### PR TITLE
fix #3239 odf2xliff doesn't extract all the text (now with unit test)

### LINKED ISSUE #3239: odf2xliff doesn't extract all the text
i use 1.12 (had problems installing 1.13rc on osx)

```
odf2xliff test.odt test.xlf
```
creates xliff files with only one translation unit containing the title (Chile)
test.odt: http://nekaka.com/d/6wo0qeVi8O
created by libreoffice 4.1.2.3  (win)

### PR DESCRIPTION
**Pull request by shaforostoff**

### CODE REVIEW

**Comment by shaforostoff** [comment]
should i create a new pull request with whitespace fixes? i'll create a new commit from scratch and push it with --force option

**Comment by unho** [comment]
@shaforostoff Please do the required changes for Travis to pass. It is only coding style issues.

File: translate/storage/xml_extract/extract.py [hunk 1]
```diff
@@ -364,20 +364,21 @@ def _walk_idml_translatable_tree(translatables, store_adder,
                                      new_parent_translatable)
 
 
-def _walk_translatable_tree(translatables, store_adder, parent_translatable):
+def _walk_translatable_tree(translatables, store_adder, parent_translatable,
+                            stored_by_parent = False):
```
Comment (unho) [inline translate/storage/xml_extract/extract.py]: No whitespace around `=`

File: translate/storage/xml_extract/extract.py [hunk 2]
```diff
@@ -364,20 +364,21 @@ def _walk_idml_translatable_tree(translatables, store_adder,
...
     """Traverse all the found translatables and add them to the Store.
 
     Inline translatables are not added to the Store.
     """
     for translatable in translatables:
-        if translatable.has_translatable_text and not translatable.is_inline:
+        store_here = (translatable.has_translatable_text and
+                    (not translatable.is_inline or not stored_by_parent))
```
Comment (unho) [inline translate/storage/xml_extract/extract.py]: Properly indent this line.

File: tests/odf_xliff/test_odf_xliff.py [hunk 3]
```diff
@@ -146,13 +148,22 @@ def test_roundtrip():
...
 
 
+def test_odf2xliff2_inline():
+    reference_xlf = factory.getobject(REFERENCE_XLF_INLINE)
+
+    odf2xliff.main(args(SOURCE_ODF_INLINE, GENERATED_XLF_TOOLKIT_INLINE))
+    generated_xlf_toolkit = factory.getobject(GENERATED_XLF_TOOLKIT_INLINE)
+    print_diff(reference_xlf, generated_xlf_toolkit)
+    assert reference_xlf == generated_xlf_toolkit
+
```
Comment (unho) [inline tests/odf_xliff/test_odf_xliff.py]: Please add another blank line. We separate functions using 2 blank lines.

File: tests/odf_xliff/test_odf_xliff.py [hunk 4]
```diff
@@ -74,6 +74,9 @@ def print_diff(store1, store2):
 REFERENCE_ODF = u'test_2.odt'
 GENERATED_ODF = u'test_2-test_roundtrip-generated.odt'
 
+SOURCE_ODF_INLINE = u'test_inline.odt'
+REFERENCE_XLF_INLINE = u'test_inline-test_odf2xliff_inline-reference.xlf'
+GENERATED_XLF_TOOLKIT_INLINE = u'test_inline-test_odf2xliff_inline-toolkit.xlf'
```
Comment (unho) [inline tests/odf_xliff/test_odf_xliff.py]: Please add another blank line here.

Comment (shaforostoff) [inline tests/odf_xliff/test_odf_xliff.py]: yeah, i already did that. i'm preparing a new commit based on master branch

**Comment by unho** [comment]
Closing this in favor of https://github.com/translate/translate/pull/3439 which includes the same changes, but without coding style issues.

**Pull request closed without merging**